File: Java/Fizz_Buzz/fizz_buzz.java
import java.util.Scanner;

class FizzBuzz {
  private static String fizzBuzz(Integer inputnum) {
    String fizzBuzz;
    if (inputnum % 15 == 0) {
      fizzBuzz = "FizzBuzz";
    } else if (inputnum % 3 == 0) {
      fizzBuzz = "Fizz";
    } else if (inputnum % 5 == 0) {
      fizzBuzz = "Buzz";
    } else {
      fizzBuzz = Integer.toString(inputnum);
    }
    return fizzBuzz;
  }

  private static void displayFizzBuzz(Integer inputnum) {
    for (int i = 1; i <= inputnum; i++) {
      System.out.print(i == inputnum ? fizzBuzz(i) : fizzBuzz(i) + ", ");
    }
  }

  private static Integer inputNum() {
    System.out.println("数字を入力してください");
    Scanner inputnum = new Scanner(System.in);
    return Integer.parseInt(inputnum.next());
  }

  public static void main(String[] args) {
    Integer inputnum = inputNum();
    displayFizzBuzz(inputnum);
  }
}

File: TypeScript/Fizz_Buzz/fizz_buzz.ts
import * as readline from "node:readline";

function fizzBuzz(n: number): string {
  if (n % 15 === 0) return "FizzBuzz";
  if (n % 3 === 0) return "Fizz";
  if (n % 5 === 0) return "Buzz";
  return String(n);
}

function joinFizzBuzz(n: number): string {
  const parts: string[] = [];
  for (let i = 1; i <= n; i++) parts.push(fizzBuzz(i));
  return parts.join(", ");
}

function question(rl: readline.Interface, prompt: string): Promise<string> {
  return new Promise((resolve) => rl.question(prompt, resolve));
}

async function promptPositiveInt(rl: readline.Interface): Promise<number> {
  const answer = (await question(rl, "数字を入力してください (1以上): ")).trim();
  const n = Number.parseInt(answer, 10);
  if (Number.isFinite(n) && n >= 1) return n;
  return promptPositiveInt(rl);
}

async function main(): Promise<void> {
  const rl = readline.createInterface({ input: process.stdin, output: process.stdout });
  try {
    const n = await promptPositiveInt(rl);
    console.log(joinFizzBuzz(n));
  } finally {
    rl.close();
  }
}

main().catch((err) => {
  console.error(err);
  process.exitCode = 1;
});

